test/conformance/api/v1/revision_test.go

```diff
@@ -0,0 +1,73 @@
+// +build e2e
+
+/*
+Copyright 2021 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import (
+	"testing"
+
+	"knative.dev/serving/test"
+	v1test "knative.dev/serving/test/v1"
+)
+
+// Test Revision Get and List operations.
+//   This test doesn't validate the Data Plane, it is just to check the Control Plane resources and their APIs
+func TestRevisionGetAndList(t *testing.T) {
+	t.Parallel()
+	clients := test.Setup(t)
+
+	names := test.ResourceNames{
+		Service: test.ObjectNameForTest(t),
+		Image:   test.PizzaPlanet1,
+	}
+
+	// Clean up on test failure or interrupt
+	test.EnsureTearDown(t, clients, &names)
+
+	// Setup initial Service
+	if _, err := v1test.CreateServiceReady(t, clients, &names); err != nil {
+		t.Fatalf("Failed to create initial Service %v: %v", names.Service, err)
+	}
+
+	// Validate State after Creation
+	if err := validateControlPlane(t, clients, names, "1"); err != nil {
+		t.Error(err)
+	}
+
+	revision, err := v1test.GetRevision(clients, names.Revision)
+	if err != nil {
+		t.Fatal("Getting revision failed")
+	}
+
+	revisions, err := v1test.GetRevisions(clients)
+	if err != nil {
+		t.Fatal("Getting revisions failed")
+	}
+	var revisionFound = false
+	for _, revisionItem := range revisions.Items {
+		t.Logf("Revision Returned: %s", revisionItem.Name)
+		if revisionItem.Name == revision.Name {
+			revisionFound = true
+		}
+	}
```

Comment:
Any reason we're not adding this to the Service test? Maybe make that `TestResourceGetAndList` and fetch Service, Configuration, Route and Revision in one shot?

Reply:
@markusthoemmes the only reason why I've decided to go in a separate file is to be able to make it simple to find the tests based on how the spec is divided into sections. If we add all the tests under `service.go` and for some reason, we want to separate those tests under different profiles, we will need again someone else who understands where the tests are for each resource and then to split them. A big part of reviewing the spec and then linking to the tests is finding out where resources are used and where they are tested. 

I did notice that there is a function to get all the resources associated with a service, that function is using Get, and List is not used anywhere. I wanted to be explicit in the test about testing GET and LIST there and make sure that the scope of the tests is just that, hence the right place for the test to live IMO should be a file called `revision_test.go`. 

What's your opinion about that? As I am coming fresh to the conformance arena, I am happy to move the test where more experienced members see fit.

Reply:
I'm fine with these smaller focused tests for now